internal/http/services/owncloud/ocdav/prop/prop.go

```diff
@@ -58,13 +59,78 @@ func EscapedNS(namespace string, local string, val string) PropertyXML {
 	}
 }
 
-// Escaped returns a new PropertyXML instance while xml-escaping the value
+var (
+	escAmp  = []byte("&amp;")
+	escLT   = []byte("&lt;")
+	escGT   = []byte("&gt;")
+	escFFFD = []byte("\uFFFD") // Unicode replacement character
+)
+
+// Decide whether the given rune is in the XML Character Range, per
+// the Char production of https://www.xml.com/axml/testaxml.htm,
+// Section 2.2 Characters.
+func isInCharacterRange(r rune) (inrange bool) {
+	return r == 0x09 ||
+		r == 0x0A ||
+		r == 0x0D ||
+		r >= 0x20 && r <= 0xD7FF ||
+		r >= 0xE000 && r <= 0xFFFD ||
+		r >= 0x10000 && r <= 0x10FFFF
+}
+
+// Escaped returns a new PropertyXML instance while replacing only
+// * `&` with `&amp;`
+// * `<` with `&lt;`
+// * `>` with `&gt;`
+// as defined in https://www.w3.org/TR/REC-xml/#syntax:
+//
+// > The ampersand character (&) and the left angle bracket (<) must not appear
+// > in their literal form, except when used as markup delimiters, or within a
+// > comment, a processing instruction, or a CDATA section. If they are needed
+// > elsewhere, they must be escaped using either numeric character references
+// > or the strings " &amp; " and " &lt; " respectively. The right angle
+// > bracket (>) may be represented using the string " &gt; ", and must, for
+// > compatibility, be escaped using either " &gt; " or a character reference
+// > when it appears in the string " ]]> " in content, when that string is not
+// > marking the end of a CDATA section.
+//
+// The code ignores errors as the legacy Escaped() does
 // TODO properly use the space
 func Escaped(key, val string) PropertyXML {
+	w := new(bytes.Buffer)
+	s := []byte(val)
+	var esc []byte
+	last := 0
+	for i := 0; i < len(s); {
+		r, width := utf8.DecodeRune(s[i:])
+		i += width
+		switch r {
+		case '&':
+			esc = escAmp
+		case '<':
+			esc = escLT
+		case '>':
+			esc = escGT
+		default:
+			if !isInCharacterRange(r) || (r == 0xFFFD && width == 1) {
```

Comment:
Nit: Would you mind changing the 0xFFFD to the utf.RuneError  constant? Same for the escFFFD variable defined above.